new-website/src/components/location/location-table.jsx

```diff
@@ -2,51 +2,91 @@ import React, { useState } from "react"
+    } else {
+        // TODO: Depending on the context we may want users to choose and not just
+        // take the first in the sequence.
+        if (userPartnerIds.length > 0) {
+            return userPartnerIds[0]
+        }
+        // TODO: Issue this error to sentry?
+        throw new Error("partnerId/unavailable")
     }
-    return userPartnerIds[0]
 }
 
-function addNewRow(userPartnerIds, onError) {
-    return async function add(data) {
-        if (!userPartnerIds || userPartnerIds.length <= 0) {
-            onError({ code: "no-permission" })
-            throw new Error("Missing Partner Id")
-        }
-        const { title, partnerId } = data
-        if (!title || title.length < 0) {
-            onError({ code: "requiredFieldMissing/title" })
-            throw new Error("missing title")
+function addNewRow(userPartnerIds, allPartnerConfigs, reportError) {
+    return async (event) => {
+        try {
+            const partnerId = choosePartnerId(null, userPartnerIds)
+            const partnerConfig = allPartnerConfigs.find(p => p.id == partnerId)
+            const collection = LOCATION_COLLECTION
+
+            navigate(`/partner/locations/`, {
+                state: { partnerId, partnerConfig, collection },
+            })
+        } catch(err) {
+            console.log(err);
+            reportError({ code: err.message })
         }
 
-        return firebase
-            .firestore()
-            .collection(LOCATION_COLLECTION)
-            .add({ ...data, partnerId: choosePartnerId(partnerId, userPartnerIds) })
-            .catch(onError)
+        // Upon confirm in add page:
+        //if (!userPartnerIds || userPartnerIds.length <= 0) {
+        //    onError({ code: "no-permission" })
+        //    throw new Error("Missing Partner Id")
+        //}
+        //const { title, partnerId } = data
+        //if (!title || title.length < 0) {
+        //    onError({ code: "requiredFieldMissing/title" })
+        //    throw new Error("missing title")
+        //}
+        //
+        //return firebase
+        //    .firestore()
+        //    .collection(LOCATION_COLLECTION)
+        //    .add({ ...data, partnerId: choosePartnerId(partnerId, userPartnerIds) })
+        //    .catch(onError)
     }
 }
 
-function deleteRow(onError) {
+function deleteRow(reportError) {
     return async function deleteIt({ id }) {
-        return firebase.firestore().collection(LOCATION_COLLECTION).doc(id).delete().catch(onError)
+        return firebase.firestore().collection(LOCATION_COLLECTION).doc(id).delete().catch(reportError)
     }
 }
 
-function updateRow(onError) {
-    return async function deleteIt({ id, ...rest }) {
-        return firebase
-            .firestore()
-            .collection(LOCATION_COLLECTION)
-            .doc(id)
-            .update({ ...rest })
-            .catch(onError)
+function updateRow(userPartnerIds, allPartnerConfigs, reportError) {
+    return async function deleteIt(event, rowData) {
```

Comment:
deleteIt was probably a copy pase error :D

Reply:
😲 Ui, yes certainly gonna fix that

Reply:
Done with https://github.com/ImmunHelden/ImmunHelden.de/pull/155/commits/30c54c8e8295592324670bff738d5dda7c62fc3f